test/gst-vaapi/vpp/csc.py

```diff
@@ -18,23 +18,20 @@
 def test_default(case, csc):
   params = spec[case].copy()
   params.update(
-    csc = csc, mcsc = mapformat(csc), mformat = mapformat(params["format"]))
+    csc = csc, mcsc = mapformat(csc), mformat = mapformat(params["format"]),
+    format2 = csc, hwup_format = mapformat_hwup(params["format"]))
   params["ofile"] = get_media()._test_artifact(
     "{}_{format}_csc_{csc}_{width}x{height}"
     ".yuv".format(case, **params))
 
   call(
     "gst-launch-1.0 -vf filesrc location={source} num-buffers={frames}"
     " ! rawvideoparse format={mformat} width={width} height={height}"
+    " ! videoconvert ! video/x-raw, format={hwup_format}"
     " ! vaapipostproc format={mcsc} ! checksumsink2 file-checksum=false"
     " frame-checksum=false plane-checksum=false dump-output=true"
     " dump-location={ofile}".format(**params))
 
-  ssim = calculate_ssim(
-    params["source"], params["ofile"],
-    params["width"], params["height"], params["frames"],
-    params["format"], params["csc"])
-
-  get_media()._set_test_details(ssim = ssim)
-
-  assert sum(ssim) == 6.0, "Y, U and V values should not be changed by CSC"
+  params.setdefault(
+    "metric", dict(type = "ssim", miny = 1.0, minu = 1.0, minv = 1.0))
+  check_metric(**params)
```

Comment:
This won't work.  The `check_metric` function relies on `params["reference"]` and `params["decoded"]`.  Neither one are in this params dictionary.  You will need to change `params["ofile"]` to `params["decoded"]` and `params["reference"]` should be set to `params["source"]` if user has not already supplied it in the config (e.g. `params.setdefault("reference", params["source"])`)